pallets/inactivity-tracking/src/tests.rs

```diff
@@ -562,3 +614,273 @@ fn inactivity_tracking_is_disabled_if_current_active_collators_storage_overflows
+        );
+    });
+}
+#[test]
+fn active_chains_noting_for_current_session_works_when_activity_tracking_is_disabled_than_enabled()
+{
+    ExtBuilder.build().execute_with(|| {
+        let current_session_active_chain_record = get_active_chains_set(vec![CONTAINER_CHAIN_ID_1]);
+        assert_eq!(
+            ActiveContainerChainsForCurrentSession::<Test>::get().len(),
+            0
+        );
+        roll_to(SESSION_BLOCK_LENGTH);
+        assert_ok!(Pallet::<Test>::set_inactivity_tracking_status(
+            RuntimeOrigin::root(),
+            false
+        ));
+        <Pallet<Test> as AuthorNotingHook<AccountId>>::on_container_authors_noted(&[
+            get_active_collators(SESSION_BLOCK_LENGTH as u32),
+        ]);
+        assert_eq!(
+            ActiveContainerChainsForCurrentSession::<Test>::get().len(),
+            0
+        );
+        let disabled_tracking_status_end_block: u32 =
+            (get_max_inactive_sessions() + 2u32) * SESSION_BLOCK_LENGTH as u32;
+        roll_to(disabled_tracking_status_end_block as u64);
+        assert_ok!(Pallet::<Test>::set_inactivity_tracking_status(
+            RuntimeOrigin::root(),
+            true
+        ));
+        // Since the activity tracking is enabled we will need to wait until the start of
+        // the next session before we are able to updating ActiveContainerChainsForCurrentSession
+        <Pallet<Test> as AuthorNotingHook<AccountId>>::on_container_authors_noted(&[
+            get_active_collators(disabled_tracking_status_end_block),
+        ]);
+        assert_eq!(
+            ActiveContainerChainsForCurrentSession::<Test>::get().len(),
+            0
+        );
+        roll_to(disabled_tracking_status_end_block as u64 + SESSION_BLOCK_LENGTH);
+        <Pallet<Test> as AuthorNotingHook<AccountId>>::on_container_authors_noted(&[
+            get_active_collators(disabled_tracking_status_end_block + SESSION_BLOCK_LENGTH as u32),
+        ]);
+        assert_eq!(
+            ActiveContainerChainsForCurrentSession::<Test>::get(),
+            current_session_active_chain_record
+        );
+    });
+}
+
+#[test]
+fn inactive_chain_collators_are_correctly_processed_when_activity_tracking_is_enabled() {
+    ExtBuilder.build().execute_with(|| {
+        assert_eq!(
+            ActiveContainerChainsForCurrentSession::<Test>::get().is_empty(),
+            true
+        );
+        assert_eq!(
+            ActiveCollatorsForCurrentSession::<Test>::get().is_empty(),
+            true
+        );
+        roll_to(SESSION_BLOCK_LENGTH - 1);
+        assert_eq!(
+            ActiveContainerChainsForCurrentSession::<Test>::get().is_empty(),
+            true
+        );
+        assert_eq!(
+            ActiveCollatorsForCurrentSession::<Test>::get().is_empty(),
+            true
+        );
+        roll_to(SESSION_BLOCK_LENGTH);
+        // Since we have one container chain with id CONTAINER_CHAIN_ID_3 which is a parathread
```

Comment:
Wait should not this be the opposite? i.e.:

- if assigned to a parathread, we dont care if a collator is active or inactive
- if assigned to a parachain, we acre to see if a collator is active or inactive

Reply:
@girazoki with the current setup we only use the filter when tracking chain inactivity (so for parathreads we ignore inactivty tracking). Do we want to ignore all parathread activity?

Reply:
Updated the code and added a explanation in https://github.com/moondance-labs/tanssi/pull/899/commits/85a78a344f8b3e1ba3574fa30c1e6eb23a95cde8